openedx/core/djangoapps/user_authn/views/password_reset.py

```diff
@@ -610,7 +612,8 @@ def password_change_request_handler(request):
     if email:
         try:
             request_password_change(email, request.is_secure())
-            user = user if user.is_authenticated else _get_user_from_email(email=email)
+            user = user if not request.POST.get('email_from_support_tools') and user.is_authenticated \
```

Comment:
which user is being authenticated here?

Reply:
there are two cases of it as you can see in the code above. Either it tries to reset password for the logged in user first or resets password for the incoming email. If the user is logged in, it'd delete it's token.